### PR TITLE
Packages: Import i18n-calypso into Calypso, take 3

### PR DESCRIPTION
**Pull request by blowery**
Take three on importing i18n-calypso into calypso's monorepo.

Previously #27123, closes #27489 (supersedes)

Very much in progress

- [x] Import existing git commits from i18n-calypso with flatten
- [ ] Migrate to ES6 / Imports
- [ ] Migrate tests to jest
- [ ] Migrate structure to src / dist to match builder
- [ ] Push new devDeps to root, if necessary
- [ ] Match up deps with calypso root
- [ ] Squash packages-prep commits
- [ ] Merge (no squash)

### CODE REVIEW

**Comment by sirreal** [comment]
Rebased and adding a few changes

File: packages/i18n-calypso/test/cli/examples/i18n-test-examples.jsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+	// Should catch template literals
+	content = i18n.translate( `My hat has six corners.` );
+	content = i18n.translate( `My hat
+has seventeen
+corners.` );
+
+	// providing context
+	content = i18n.translate( {
+		original: 'post',
```
Comment (a8ci18n) [inline packages/i18n-calypso/test/cli/examples/i18n-test-examples.jsx]: :information_source: This string already exists with the following **contexts**:
- `null` (no context)
- `add new from admin bar`
- `noun`
- `post a comment`
- `post type singular name`
- `Verb, to post`

 Would it make sense to reuse one of the above?

File: packages/i18n-calypso/test/cli/examples/i18n-test-examples.jsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+		context: 'verb'
+	} );
+
+	// passing string as initial argument
+	content = i18n.translate( 'post2', { context: 'verb2' } );
```
Comment (a8ci18n) [inline packages/i18n-calypso/test/cli/examples/i18n-test-examples.jsx]: :information_source: This string doesn't exist elsewhere without a **context** (or with a different one). You can probably drop the context. If needed, you can use a translator comment instead to clarify meaning.

File: packages/i18n-calypso/test/cli/examples/i18n-test-examples.jsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	var varCount = 1;
+	content = i18n.translate( 'single test2', 'plural test2', { count: varCount } );
+
+	// test the ability to break strings to multiple lines
+	content = <div>{ i18n.translate(
+		"This is a multi-line translation with \"mixed quotes\" " +
+		'and mixed \'single quotes\''
+	) }</div>;
+
+	content = this.translate( 'The string key text',
```
Comment (a8ci18n) [inline packages/i18n-calypso/test/cli/examples/i18n-test-examples.jsx]: :information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

File: packages/i18n-calypso/test/index.js [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+			} );
+			it( 'should fall back to original if translation isn\'t even null in locale file', function() {
+				assert.equal( 'nonexisting-string', translate( 'nonexisting-string' ) );
+			} );
+		} );
+
+		describe( 'translate with context', function() {
+			it( 'should find a string with context', function() {
+				assert.equal( 'translation3', translate( {
+					original: 'test3',
```
Comment (a8ci18n) [inline packages/i18n-calypso/test/index.js]: :information_source: This string doesn't exist elsewhere without a **context** (or with a different one). You can probably drop the context. If needed, you can use a translator comment instead to clarify meaning.

File: packages/i18n-calypso/package.json [hunk 5]
```diff
@@ -1,48 +1,45 @@
...
-    "rewire": "^2.5.1"
-  }
+	"name": "i18n-calypso",
+	"version": "3.0.0",
+	"description": "i18n JavaScript library on top of Jed originally used in Calypso",
+	"main": "dist/cjs/index.js",
+	"module": "dist/esm/index.js",
+	"repository": {
+		"type": "git",
+		"url": "git+https://github.com/Automattic/i18n-calypso.git"
```
Comment (blowery) [inline packages/i18n-calypso/package.json]: needs an update

File: packages/i18n-calypso/package.json [hunk 6]
```diff
@@ -1,48 +1,45 @@
...
+	},
+	"author": "@automattic",
+	"license": "GPL-2.0",
+	"bin": {
+		"i18n-calypso": "bin/index.js"
+	},
+	"bugs": {
+		"url": "https://github.com/Automattic/i18n-calypso/issues"
```
Comment (blowery) [inline packages/i18n-calypso/package.json]: update

File: packages/i18n-calypso/package.json [hunk 7]
```diff
@@ -1,48 +1,45 @@
...
+	},
+	"homepage": "https://github.com/Automattic/i18n-calypso#readme",
```
Comment (blowery) [inline packages/i18n-calypso/package.json]: update

**Comment by sirreal** [comment]
i18n job should be fixed by #29260 
I'd like to ignore the tests as well so this can be merged preserving the history. Initially, only the source would move into Calypso without being used directly, running tests, or any significant impact on Calypso.

The cleanup and prep can occur in #29229 and other follow-ups. There's a significant amount for this particular package.

**Comment by sirreal** [comment]
I've disabled tests and cherry-picked the i18n fix. Any objections to landing this and doing iterative cleanup in follow-ups?

**Review by sirreal — APPROVED** [review]

**Comment by sirreal** [comment]
Rebased, i18n fix is now on master.

**Comment by blowery** [comment]
Looks like this is still symlinking in i18n-calypso and breaking calypso.

**Comment by blowery** [comment]
I think if we want to land this, we'll need to temporarily change the package name to prevent lerna from symlinking it to calypso.

**Comment by sirreal** [comment]
> I think if we want to land this, we'll need to temporarily change the package name to prevent lerna from symlinking it to calypso.

Gave that a spin in 87801923ccd39f43848306c66121fd1ea1737c28

**Comment by blowery** [comment]
@sirreal ` Error: Invalid package name "__i18n-calypso": name cannot start with an underscore`

File: lerna.json [hunk 8]
```diff
@@ -2,8 +2,9 @@
 	"packages": [ "packages/*", "." ],
 	"version": "independent",
 	"ignoreChanges": [
-		"**/.eslintrc.js",
 		"**/*.md",
-		"**/package-lock.json"
+		"**/.eslintrc.{js,json,yaml,yml}",
+		"**/package-lock.json",
+		"packages/i18n-calypso"
```
Comment (sirreal) [inline lerna.json]: Fix in follow-up

File: packages/i18n-calypso/package.json [hunk 9]
```diff
@@ -0,0 +1,48 @@
+{
+  "name": "not-i18n-calypso-itreallyis",
```
Comment (sirreal) [inline packages/i18n-calypso/package.json]: Fix in follow-up

Comment (blowery) [reply packages/i18n-calypso/package.json]: nice. :)

might need to run update-deps too?

File: packages/i18n-calypso/_test/cli/examples/i18n-test-examples.jsx [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+
+	// Should catch template literals
+	content = i18n.translate( `My hat has six corners.` );
+	content = i18n.translate( `My hat
+has seventeen
+corners.` );
+
+	// providing context
+	content = i18n.translate( {
+		original: 'post',
```
Comment (a8ci18n) [inline packages/i18n-calypso/_test/cli/examples/i18n-test-examples.jsx]: :information_source: This string already exists with the following **contexts**:
- `null` (no context)
- `add new from admin bar`
- `noun`
- `post a comment`
- `post type singular name`
- `Verb, to post`

 Would it make sense to reuse one of the above?

File: packages/i18n-calypso/_test/cli/examples/i18n-test-examples.jsx [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+		context: 'verb'
+	} );
+
+	// passing string as initial argument
+	content = i18n.translate( 'post2', { context: 'verb2' } );
```
Comment (a8ci18n) [inline packages/i18n-calypso/_test/cli/examples/i18n-test-examples.jsx]: :information_source: This string doesn't exist elsewhere without a **context** (or with a different one). You can probably drop the context. If needed, you can use a translator comment instead to clarify meaning.

File: packages/i18n-calypso/_test/cli/examples/i18n-test-examples.jsx [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+	var varCount = 1;
+	content = i18n.translate( 'single test2', 'plural test2', { count: varCount } );
+
+	// test the ability to break strings to multiple lines
+	content = <div>{ i18n.translate(
+		"This is a multi-line translation with \"mixed quotes\" " +
+		'and mixed \'single quotes\''
+	) }</div>;
+
+	content = this.translate( 'The string key text',
```
Comment (a8ci18n) [inline packages/i18n-calypso/_test/cli/examples/i18n-test-examples.jsx]: :information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

File: packages/i18n-calypso/_test/index.js [hunk 13]
```diff
@@ -0,0 +1,366 @@
...
+			} );
+			it( 'should fall back to original if translation isn\'t even null in locale file', function() {
+				assert.equal( 'nonexisting-string', translate( 'nonexisting-string' ) );
+			} );
+		} );
+
+		describe( 'translate with context', function() {
+			it( 'should find a string with context', function() {
+				assert.equal( 'translation3', translate( {
+					original: 'test3',
```
Comment (a8ci18n) [inline packages/i18n-calypso/_test/index.js]: :information_source: This string doesn't exist elsewhere without a **context** (or with a different one). You can probably drop the context. If needed, you can use a translator comment instead to clarify meaning.

**Comment by blowery** [comment]
do we need to ignore the translations in the test directory?

**Comment by sirreal** [comment]
> do we need to ignore the translations in the test directory?

That's probably a good idea

File: package.json [hunk 14]
```diff
@@ -271,7 +271,7 @@
     "test-server": "jest -c=test/server/jest.config.js",
     "test-server:coverage": "npm run -s test-server -- --coverage",
     "test-server:watch": "npm run -s test-server -- --watch",
-    "translate": "i18n-calypso --format pot --output-file ./calypso-strings.pot -k translate,__,_x,_n,_nx -e date \"**/*.js\" \"**/*.jsx\" \"!build/**\" \"!public/**\" \"!node_modules/**\" \"!**/node_modules/**\"",
+    "translate": "i18n-calypso --format pot --output-file ./calypso-strings.pot -k translate,__,_x,_n,_nx -e date \"**/*.js\" \"**/*.jsx\" \"!build/**\" \"!public/**\" \"!node_modules/**\" \"!**/node_modules/**\" \"!packages/i18n-calypso/_test/**\" # FIXME: when tests are restored",
```
Comment (sirreal) [inline package.json]: Update when restoring tests

**Review by sirreal — APPROVED** [review]
Tests and full e2e are all green. This shouldn't modify any Calypso behavior. Let's 🚢!

**Comment by ockham** [comment]
:clap: Thanks for bringing this home, folks!

**Comment by sirreal** [comment]
> 👏 Thanks for bringing this home, folks!

It's not quite home yet, need to land #29229, probably in a series of 2-3 incremental PRs

**Pull request merged**